stake-pool/birdbath/src/processor.rs

```diff
@@ -0,0 +1,1100 @@
+        // i just like it this way because no account validation required lol
+
+        // create the pool mint
+        let mint_space = spl_token::state::Mint::LEN;
+        let mint_rent = rent.minimum_balance(mint_space);
+
+        invoke_signed(
+            &system_instruction::create_account(
+                payer_info.key,
+                pool_mint_info.key,
+                mint_rent,
+                mint_space as u64,
+                token_program_info.key,
+            ),
+            &[
+                payer_info.clone(),
+                pool_mint_info.clone(),
+                system_program_info.clone(),
+            ],
+            mint_signers,
+        )?;
+
+        invoke_signed(
+            &spl_token::instruction::initialize_mint(
+                token_program_info.key,
+                pool_mint_info.key,
+                pool_authority_info.key,
+                None,
+                MINT_DECIMALS,
+            )?,
+            &[
+                pool_mint_info.clone(),
+                rent_info.clone(),
+                system_program_info.clone(),
+            ],
+            authority_signers,
+        )?;
+
+        // create the pool stake account
+        let stake_space = std::mem::size_of::<stake::state::StakeState>();
+        let required_lamports = rent.minimum_balance(stake_space).saturating_add(1);
+        let authorized = stake::state::Authorized::auto(pool_authority_info.key);
+
+        invoke_signed(
+            &system_instruction::create_account(
+                payer_info.key,
+                pool_stake_info.key,
+                required_lamports,
+                stake_space as u64,
+                stake_program_info.key,
+            ),
+            &[
+                payer_info.clone(),
+                pool_stake_info.clone(),
+                stake_program_info.clone(),
+            ],
+            stake_signers,
+        )?;
+
+        invoke_signed(
+            &stake::instruction::initialize_checked(pool_stake_info.key, &authorized),
+            &[
+                pool_stake_info.clone(),
+                rent_info.clone(),
+                pool_authority_info.clone(),
+                pool_authority_info.clone(),
+            ],
+            authority_signers,
+        )?;
+
+        // delegate the stake so it activates
+        invoke_signed(
+            &stake::instruction::delegate_stake(
+                pool_stake_info.key,
+                pool_authority_info.key,
+                validator_vote_info.key,
+            ),
+            &[
+                pool_stake_info.clone(),
+                validator_vote_info.clone(),
+                clock_info.clone(),
+                stake_history_info.clone(),
+                stake_config_info.clone(),
+                pool_authority_info.clone(),
+            ],
+            authority_signers,
+        )?;
+
+        // could mint the token here if we wanted, either to user or incinerator
```

Comment:
It would be economically sound to give pool tokens for the stake account minimum stake, at the very least, since they'll be giving rent-exemption to make the mint and stake account.

Reply:
oh i only take 1 lamport from the initializer and assume theyll deposit more normally. i kinda like the edge case protection that theres always one lamport unspoken for by any tokens in circulation